### PR TITLE
Write Version selector data to Registry

### PR DESCRIPTION
**Pull request by sharadkjaiswal**
### Purpose
- Fixes MAGN-8747: Default Dynamo version should 0.9.0 once it is installed.

### Problem
 Earlier the version selector data was saved in %appdata%\DynamoDllForLoad_2014.txt file. So when the newer version of Dynamo is installed, the version selector finds this file and reads the last version and shows up in the split button. We either need to delete this file at the time of installation or keep track of first run after the installation.

### Solution
  Instead of using local file to keep version specific data, I moved this data to registry of the current Dynamo version. Version selector can read this information from registry when the version selector app is loaded in Revit and update the Default Dynamo version in split button accordingly.
  Defined a new class VersionSelectorData to keep the selected version data to be stored in registry under Dynamo product information key. When product gets uninstalled, the registry data will automatically be cleaned.

### Declarations

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@Randy-Ma

### CODE REVIEW

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs [hunk 1]
```diff
@@ -54,18 +55,9 @@ public Result OnStartup(UIControlledApplication application)
             uiApplication = application;
             // now we have a default path, but let's look at
             // the load path file to see what was last selected
```
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: Minor, but the comments need to be updated.

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs [hunk 2]
```diff
@@ -358,45 +353,72 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
...
-            if (File.Exists(path))
-            {
-                File.Delete(path);
-            }
-
-            using (var tw = new StreamWriter(path))
-            {
-                tw.WriteLine(loadPath);
-            }
+        const string RegKey64 = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\";
```
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: I think we need a different location.

Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: I see, the data stored here are going to be cleaned after uninstallation.

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs [hunk 3]
```diff
@@ -74,18 +66,22 @@ public Result OnStartup(UIControlledApplication application)
...
             int index = -1;
+            var selectedVersion = selectorData.SelectedVersion.ToString(2);
             foreach (var p in dynamoProducts)
             {
                 var path = VersionLoader.GetDynamoRevitPath(p, revitVersion);
                 if (!File.Exists(path))
                     continue;
 
-                if (path.Equals(cachedPath))
+                if (p.VersionInfo.ToString(2) == selectedVersion)
```
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: It is better to make it consistent to take 3 fields like what is done in the AddSplitButtonGroup function. Or is this done for a reason?

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs [hunk 4]
```diff
@@ -358,45 +353,72 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
...
         {
-            try
-            {
-                string appDataFolder = System.Environment.GetFolderPath(
-                    System.Environment.SpecialFolder.ApplicationData);
+            var data = new VersionSelectorData() { RevitVersion = revitVersion };
+            var regKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry64);
+            regKey = regKey.OpenSubKey(RegKey64);
+            
+            var key = string.Format(@"Dynamo {0}.{1}", data.ThisDynamoVersion.Major, data.ThisDynamoVersion.Minor);
```
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: What is the benefit for a structure like:
```
Dynamo {major version}.{minor version}
|----{RevitVersion}
|----{RevitVersion}
```
There might be one entry every time a different version of Dynamo is installed.

**Comment by Randy-Ma** [comment]
This is great! The version selector is more flexible than before. One thing to check is after a new version is installed, we'd better remove the obsoleted entries in the registry. Apart from this and few comments, LGTM

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs (hunk 3 above)
Comment (sharadkjaiswal) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: Currently, we can't have 0.9.0 and 0.9.1 installed side-by-side, hence no point comparing the 3rd field.

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs (hunk 4 above)
Comment (sharadkjaiswal) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: Agree, there will be a different entry for every version of Dynamo installed, but these will be removed when Dynamo gets uninstalled.

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs (hunk 3 above)
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: Thanks! This makes sense!

File: src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs (hunk 4 above)
Comment (Randy-Ma) [inline src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs]: Yes, thanks

**Pull request merged**